src/drivers/glfw/lv_glfw_window.c

```diff
@@ -89,6 +103,19 @@ lv_glfw_window_t * lv_glfw_window_create(int32_t hor_res, int32_t ver_res, bool
         return NULL;
     }
 
+    title = (window_title != NULL) ? (char *)window_title : (char *)default_title;
+    if(desktop_file_name == NULL) {
+        char buffer[256];
+        lv_snprintf(buffer, 255, "%s %s", title, default_suffix);
+        buffer[255] = '\0';
```

Comment:
snprintf always terminates buffers

```suggestion
        char buffer[256];
        lv_snprintf(buffer, sizeof(buffer), "%s %s", title, default_suffix);
```

Reply:
My research suggests this is not always true, here's what I found:

first, snprintf() does work like you're saying... but microsofts "_snprintf()" (note, with underscore), does not:

[](https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2012/2ts7cx93(v=vs.110)?redirectedfrom=MSDN)

If len < count, then len characters are stored in buffer, a null-terminator is appended, and len is returned.

If len = count, then len characters are stored in buffer, **no null-terminator is appended**, and len is returned.

If len > count, then count characters are stored in buffer, **no null-terminator is appended**, and a negative value is returned.

and then there's this, which makes me think it's a good safety feature to have in there, even if it's usually redundant:
[](https://nvd.nist.gov/vuln/detail/CVE-2018-1000101)

Mingw-w64 version 5.0.3 and earlier, 5.0.4, 6.0.0 and 7.0.0 contains an Improper Null Termination (CWE-170) vulnerability in mingw-w64-crt (libc)->(v)snprintf that can result in The bug may be used to corrupt subsequent string functions. This attack appear to be exploitable via Depending on the usage, worst case: network.

That appears to be because those versions use the microsoft underscore version ("_snprintf()").

So I'd prefer we keep that explicit null termination in there, just to be absolutely sure we aren't introducing a possible major security exploit into anyone else's code utilizing this.  I realize this probably seems overly cautious but the stakes are very high for something like this, the way LVGL gets implemented in so many other projects.  I would hate to be the one that let an exploit slip through.

Also, if lv_snprintf() does not explicitly null terminate, it seems like it should.